web/libs/core/src/pages/AccountSettings/sections/PersonalInfo.tsx

```diff
@@ -1,90 +1,139 @@
+      },
+      {
+        body,
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+        errorFilter: () => true,
+      },
+    );
+    return response;
+  },
+}));
 
 export const PersonalInfo = () => {
   const api = useAPI();
   const toast = useToast();
-  const { user, fetch, isInProgress: userInProgress } = useCurrentUser();
-  const [fname, setFName] = useState("");
-  const [lname, setLName] = useState("");
-  const [email, setEmail] = useState("");
-  const [phone, setPhone] = useState("");
+  const updateUser = useAtomValue(updateUserAtom);
+  const updateUserAvatar = useAtomValue(updateUserAvatarAtom);
+  const { user, fetch: refetchUser, isInProgress: userInProgress } = useCurrentUser();
   const [isInProgress, setIsInProgress] = useState(false);
-  const userInfoForm = useRef();
-  const userAvatarForm = useRef();
-  const avatarRef = useRef();
-  const fileChangeHandler = () => userAvatarForm.current.requestSubmit();
+  const userAvatarForm = useRef<HTMLFormElement>();
+  const avatarRef = useRef<HTMLInputElement>();
+  const fileChangeHandler: FormEventHandler<HTMLInputElement> = useCallback(
+    async (e) => {
+      if (!user) return;
+
+      const input = e.currentTarget as HTMLInputElement;
+      const body = new FormData();
+      body.append("avatar", input.files?.[0] ?? "");
+      const response = await updateUserAvatar.mutateAsync({
+        body,
+        userId: user.id,
+      });
+
+      if (!response.$meta.ok) {
+        toast.show({ message: response?.response?.detail ?? "Error updating avatar", type: ToastType.error });
+      } else {
+        refetchUser();
+      }
+      input.value = "";
+    },
+    [user?.id],
+  );
+
+  const deleteUserAvatar = async () => {
+    if (!user) return;
+    await updateUserAvatar.mutateAsync({ userId: user.id, isDelete: true });
+    refetchUser();
+  };
+
   const avatarFormSubmitHandler = useCallback(
-    async (e, isDelete = false) => {
+    async (e: FormEvent, isDelete = false) => {
       e.preventDefault();
-      const response = await api.callApi(isDelete ? "deleteUserAvatar" : "updateUserAvatar", {
-        params: {
-          pk: user?.id,
-        },
-        body: {
-          avatar: avatarRef.current.files[0],
-        },
-        headers: {
-          "Content-Type": "multipart/form-data",
-        },
-        errorFilter: () => true,
+      e.stopPropagation();
+      if (!user) return;
+
+      const body = new FormData(e.currentTarget as HTMLFormElement);
+      const response = await updateUserAvatar.mutateAsync({
+        body,
+        userId: user.id,
       });
-      if (!isDelete && response?.status) {
+
+      if (!isDelete && !response.$meta.ok) {
         toast.show({ message: response?.response?.detail ?? "Error updating avatar", type: "error" });
       } else {
-        fetch();
+        refetchUser();
       }
-      userAvatarForm.current.reset();
+      userAvatarForm.current?.reset();
```

Comment:
should not we also use it in `fileChangeHandler()`?

Reply:
that's actually an unused function, removed it